UnitySetup/UnitySetup.psm1

```diff
@@ -1870,7 +1870,22 @@ function Start-UnityEditor {
             $process.Start() | Out-Null
 
             if ( $Wait ) {
-                $process.WaitForExit()
+                if( $LogFile ) {
+                    New-Item -Path $LogFile -ItemType File -Force
+                    $ljob = Start-Job -ScriptBlock { param($log) Get-Content "$log" -Wait } -ArgumentList $LogFile
+
+                    while ( -not $process.HasExited -and $ljob.HasMoreData )
```

Comment:
Should this just loop on `-not $process.HasExited` and then just call Receive-Job inside the loop? Otherwise if there's not data, but the process hasn't exited we'll quit the loop and exit before the process is complete.

Reply:
I think as long as unity is writing to that file, there is always more data. 

The real problem I'm having is that there is more data but the process exits before I can get it all printed to the console lol. But I don't really mind as we get the full log in the artifacts anyway.

Reply:
Your call to Receive-Job below should pass -Wait so that it retrieves all the results of the job. That should ensure you get all the contents of the file.

Re: always having more data - again a bit of dependency on Unity spamming and a race condition between the `Get-Content` job and Unity. Better to be explicit and just loop until the process has exited - you can blindly make calls to `Receive-Job` in the loop, then do the `Receive-Job -Wait -AutoRemove` after the loop.

Reply:
Before when I just waited for `hasMoreData` the loop would just hang forever, even when the unity process quit. Wasn't sure if this would have the same result, but I'll let you know once the CI runs on my end.